lib/IdP/SAML2.php

```diff
@@ -306,42 +319,54 @@ public static function receiveAuthnRequest(SimpleSAML_IdP $idp)
                 $nameIDFormat = null;
             }
 
+            if (isset($_REQUEST['ConsumerURL'])) {
+                $consumerURL = (string)$_REQUEST['ConsumerURL'];
+            } elseif (isset($_REQUEST['shire'])) {
+                $consumerURL = (string)$_REQUEST['shire'];
```

Comment:
I didn't know [the Shire](https://en.wikipedia.org/wiki/Shire_(Middle-earth)) used SAML... 😉

Reply:
Haha, now you know...